### PR TITLE
Add autoyast installation test for svirt-xen-hvm

### PR DESCRIPTION
**Pull request by lemon-suse**
This is the first time to add autoyast installation test for svirt-xen-hvm in openQA.

- Related ticket: https://progress.opensuse.org/issues/160120
- Related MR: https://gitlab.suse.de/qe-yam/openqa-job-groups/-/merge_requests/278
- Needles: N/A
- Verification run: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP6&build=93.5&groupid=316

### CODE REVIEW

**Comment by rakoenig** [comment]
LGTM.

File: data/yam/autoyast/create_hdd_textmode_xen_hvm.xml [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+        <address>2.suse.pool.ntp.org</address>
+        <iburst t="boolean">true</iburst>
+        <offline t="boolean">false</offline>
+      </ntp_server>
+    </ntp_servers>
+    <ntp_sync>manual</ntp_sync>
+  </ntp-client>
+  <partitioning t="list">
+    <drive t="map">
+      <device>/dev/xvdb</device>
```
Comment (rfan1) [inline data/yam/autoyast/create_hdd_textmode_xen_hvm.xml:100]: Is it the right disk name here?

File: tests/autoyast/installation.pm [hunk 2]
```diff
@@ -320,6 +321,12 @@ sub run {
         }
     }
 
+    # For xen, need set hard-disk as the boot device, then reboot and connect to existing VNC.
+    if (check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'hvm')) {
```
Comment (rfan1) [inline tests/autoyast/installation.pm:325]: Hmm, is it enough if only `check_var('VIRSH_VMM_TYPE', 'hvm')`

File: tests/autoyast/installation.pm [hunk 3]
```diff
@@ -127,7 +128,7 @@ sub run {
     push @needles, 'autoyast-error' unless get_var('AUTOYAST_EXPECT_ERRORS');
     # Autoyast reboot automatically without confirmation, usually assert 'bios-boot' that is not existing on zVM
     # So push a needle to check upcoming reboot on zVM that is a way to indicate the stage done
-    push @needles, 'autoyast-stage1-reboot-upcoming' if is_s390x || (is_pvm && !is_upgrade);
+    push @needles, 'autoyast-stage1-reboot-upcoming' if is_s390x || (is_pvm && !is_upgrade) || (check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'hvm'));
```
Comment (rfan1) [inline tests/autoyast/installation.pm:131]: Is it fine for setting `check_var('VIRSH_VMM_TYPE', 'hvm')` only？

File: data/yam/autoyast/create_hdd_textmode_xen_hvm.xml (hunk 1 above)
Comment (lemon-suse) [reply data/yam/autoyast/create_hdd_textmode_xen_hvm.xml:100]: Thanks. It's cloned from interactive installation, for VR this works well.

File: tests/autoyast/installation.pm (hunk 2 above)
Comment (lemon-suse) [reply tests/autoyast/installation.pm:325]: For me with these two setting together to make the condition more accuracy for xen-hvm, so I prefer to use these two together as condition. Thanks.

File: tests/autoyast/installation.pm (hunk 3 above)
Comment (lemon-suse) [reply tests/autoyast/installation.pm:131]: For me with these two setting together to make the condition more accuracy for xen-hvm, so I prefer to use these two together as condition. Thanks.

File: data/yam/autoyast/create_hdd_textmode_xen_hvm.xml (hunk 1 above)
Comment (rfan1) [reply data/yam/autoyast/create_hdd_textmode_xen_hvm.xml:100]: From serial log, it is vda 
```
[  292.152996][ T2644] BTRFS info (device vda2): balance: start -dusage=0
[  292.153778][ T2644] BTRFS info (device vda2): balance: ended with status: 0
[  292.157013][ T2645] BTRFS info (device vda2): balance: start -dusage=5
[  292.157767][ T2645] BTRFS info (device vda2): balance: ended with status: 0
[  292.160817][ T2646] BTRFS info (device vda2): balance: start -dusage=10
[  292.161593][ T2646] BTRFS info (device vda2): balance: ended with status: 0
[  292.164825][ T2647] BTRFS info (device vda2): balance: start -musage=0 -susage=0
[  292.165677][ T2647] BTRFS info (device vda2): balance: ended with status: 0
[  292.168696][ T2648] BTRFS info (device vda2): balance: start -musage=3 -susage=3
[  292.169618][ T2648] BTRFS info (device vda2): relocating block group 22020096 flags system|dup
[  292.173288][ T2648] BTRFS info (device vda2): balance: ended with status: 0
[  293.313489][ T2740] BTRFS info (device vda2): scrub: started on devid 1
[  294.805122][ T2786] BTRFS warning (device vda2): failed to trim 5 block group(s), last error -512
[  294.812640][ T2786] BTRFS warning (device vda2): failed to trim 1 device(s), last error -512
[  300.177948][  T169] BTRFS info (device vda2): qgroup scan completed (inconsistency flag cleared)
```

**Review by rfan1 — APPROVED** [review]
LGTM

File: data/yam/autoyast/create_hdd_textmode_xen_hvm.xml (hunk 1 above)
Comment (lemon-suse) [reply data/yam/autoyast/create_hdd_textmode_xen_hvm.xml:100]: It is not supported to set 'vda', https://openqa.suse.de/tests/14981365#step/installation/44

**Pull request merged**